src/components/Tool/GoogleImport/GoogleImport.tsx

```diff
@@ -0,0 +1,553 @@
+import React, { ReactElement, useEffect, useMemo, useState } from 'react';
+import {
+  Autocomplete,
+  Box,
+  Button,
+  ButtonGroup,
+  Card,
+  CardActions,
+  CardContent,
+  CardHeader,
+  Checkbox,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  Divider,
+  FormControl,
+  FormControlLabel,
+  Grid,
+  MenuItem,
+  Radio,
+  RadioGroup,
+  Select,
+  Typography,
+} from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { Formik } from 'formik';
+import { Trans, useTranslation } from 'react-i18next';
+import { makeStyles } from 'tss-react/mui';
+import * as yup from 'yup';
+import { useGetContactTagListQuery } from 'src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.generated';
+import { LoadingSpinner } from 'src/components/Settings/Organization/LoadingSpinner';
+import { ContactTagInput } from 'src/components/Tags/Tags';
+import Modal from 'src/components/common/Modal/Modal';
+import useGetAppSettings from 'src/hooks/useGetAppSettings';
+import theme from 'src/theme';
+import NoData from '../NoData';
+import { useGoogleContactGroupsQuery } from './googleContactGroups.generated';
+
+const BoldTypography = styled(Typography)(() => ({
+  fontWeight: 'bold',
+}));
+
+const ContainerBox = styled(Grid)(({ theme }) => ({
+  padding: theme.spacing(3),
+  width: '70%',
+  display: 'flex',
+  [theme.breakpoints.down('lg')]: {
+    width: '100%',
+  },
+}));
+
+const OuterBox = styled(Box)(() => ({
+  display: 'flex',
+  flexDirection: 'row',
+  justifyContent: 'center',
+  width: '100%',
+}));
+
+const HeaderBox = styled(Box)(({ theme }) => ({
+  backgroundColor: theme.palette.cruGrayLight.main,
+  width: '100%',
+  padding: theme.spacing(1),
+  display: 'flex',
+  justifyContent: 'end',
+  borderTopRightRadius: '7px',
+  borderTopLeftRadius: '7px',
+  [theme.breakpoints.down('sm')]: {
+    paddingTop: theme.spacing(2),
+  },
+}));
+
+const BorderBox = styled(Box)(({ theme }) => ({
+  marginTop: theme.spacing(2),
+  marginBottom: theme.spacing(2),
+  border: `1px solid ${theme.palette.cruGrayLight.main}`,
+  width: '100%',
+  borderRadius: '7px',
+}));
+
+const Section = styled(Box)(({ theme }) => ({
+  marginTop: theme.spacing(2),
+}));
+
+const useStyles = makeStyles()(() => ({
+  bulletList: {
+    margin: '0 0 10px 15px',
+    '> li': {
+      margin: '5px',
+    },
+  },
+  divider: {
+    margin: theme.spacing(2, 0),
+  },
+  headerTitle: {
+    fontSize: '1rem!important',
+  },
+}));
+
+const googleImportSchema = yup.object({
+  tagsForAllList: yup.array().of(yup.string()).default([]),
+  override: yup.string().required(),
+  importAll: yup.string().required(),
+  groupTags: yup.object(),
+  groups: yup.array().of(yup.string()).default([]),
+});
```

Comment:
Do you think it's worth adding error messages since all these are required.

Reply:
Most of these are radio buttons so they will always have a value. I made the button disabled for the one field they need to submit. I'll also remove .required()